tests/Rules/TagTest.php

```diff
@@ -3,70 +3,85 @@
 declare(strict_types=1);
 
 use ARKEcosystem\UserInterface\Rules\Tag;
+use ARKEcosystem\UserInterface\Support\Enums\Constants;
 
-it('accepts a simple word', function () {
-    $rule = new Tag();
-    $tag = 'hello';
-    $this->assertTrue($rule->passes('tag', $tag));
+beforeEach(function () {
```

Comment:
add a few more test cases, currently missing things like:
- spaces at the start/end
- dashes at the start/end
- spaces and dashes combined (e.g. `h-e l-l o`)
- consecutive spaces and dashes (e.g. `hi- there`)
- multiple dashes
- multiple spaces
- some more additional special characters
- some weird characters like a `three em dash` (`⸻`) that is part of the dash family (should not be allowed)